pkg/agent/proxy/proxier.go

```diff
@@ -117,13 +112,23 @@ func (p *proxier) removeStaleServices() {
 
 func (p *proxier) removeStaleEndpoints(staleEndpoints map[k8sproxy.ServicePortName]map[string]k8sproxy.Endpoint) {
 	for svcPortName, endpoints := range staleEndpoints {
-		bindingProtocol := binding.ProtocolTCP
-		if svcPortName.Protocol == corev1.ProtocolUDP {
-			bindingProtocol = binding.ProtocolUDP
-		} else if svcPortName.Protocol == corev1.ProtocolSCTP {
-			bindingProtocol = binding.ProtocolSCTP
-		}
+		var bindingProtocol binding.Protocol
 		for _, endpoint := range endpoints {
+			if utilnet.IsIPv6String(endpoint.IP()) {
```

Comment:
I suggest extracting this big if block to a separate function to improve readability. The go compiler will inline the function I guess, then there should be no different.